File: API/client-app/src/app/shared/models/part.ts
export interface IPartPagination {
  items: IPart[]
  pageNumber: number
  totalPages: number
  totalCount: number
  hasPreviousPage: boolean
  hasNextPage: boolean
}

export interface IPart {
  partNumber: string
  oracleCode: string
  localCode: string

  model: string
  description: string
  created: string
  addVoucher: number
  createdBy: string
  customer: string
  warehouse: string
  engineer: string
  orderDate: string
  partStatus: number
  brandId: number
  warehouseId: number
  customerId: number
  engneerId: number
  image: string
}

File: Domain/Entities/Part.cs
using Domain.Common;
using Domain.Enum;
using System.ComponentModel.DataAnnotations;

namespace Domain.Entities
{
    public class Part:BaseAuditableEntity
    {
        [Required]
        public string PartNumber { get; set; }
        [Required]
        public string OracleCode { get; set; }
        public string? LocalCode { get; set; }

        public ItemStatus PartStatus { get; set; } = ItemStatus.stored;
        public DateTime? OrderDate { get; set; }
        public int ExceededLimit { get; set; }
        public string Model { get; set; }
        public string? Description { get; set; }
        public string AddVoucher { get; set; }

        public int WarehouseId { get; set; }
        public Warehouse Warehouse { get; set; }
        public int? CustomerId { get; set; }
        public Customer? Customer { get; set; }
        public int? EngneerId { get; set; }
        public Engineer? Engineer { get; set; }
        public byte[]? Image { get; set; }
    }
}
